### ISSUE TITLE
upper_case_acronyms warns on public items

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#6803]
I recently ran nightly clippy on a project and got the following warning:

```
error: name `YDBError` contains a capitalized acronym
  --> src/simple_api/mod.rs:95:12
   |
95 | pub struct YDBError {
   |            ^^^^^^^^ help: consider making the acronym lowercase, except the initial letter: `YdbError`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms
```
The lint is correct, it's not a false positive. However, fixing it would be a breaking change since `YDBError` is public. Maybe this lint should be off by default since it's only a style lint and it can be hard to fix?

I originally just added `#[allow(clippy::upper_case_literals)]`, but that gives more warnings on stable until this rides the release trains unless I add `allow(unknown_renamed_lints)`, which I'd rather not do.

**Comment by matthiaskrgr** [comment]
Huh, does this still happen?
I'm not getting a warning with `rustc 1.52.0-nightly (98f8cce6d 2021-02-25)`

This should have been "fixed" by https://github.com/rust-lang/rust-clippy/pull/6788 .
Anyway the linting of public items is probably still a bug and still needs fixing (for the gated "aggressive" mode of the lint..)

You mean the `upper_case_acronyms` lint, right?

**Comment by jyn514** [comment]
> This should have been "fixed" by #6788 .

Thanks, it's a lot better after that! It still warns on variants of public enums though:

```
warning: name `YDB` contains a capitalized acronym
   --> src/context_api/mod.rs:747:5
    |
747 |     YDB(YDBError),
    |     ^^^ help: consider making the acronym lowercase, except the initial letter: `Ydb`
    |
    = note: `#[warn(clippy::upper_case_acronyms)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms
```
for the code
```rust
pub enum ParseError<T> {
    YDB(YDBError),
    Utf8(std::string::FromUtf8Error),
    Parse(T, String),
}
```

**Comment by jyn514** [comment]
@matthiaskrgr this still seems to be broken with nightly clippy:
```
$ clippy-driver --version
clippy 0.1.52 (673d0db 2021-03-23)
$ cargo clippy -- -D clippy::all
error: name `YDB` contains a capitalized acronym
 --> src/context_api/mod.rs:747:5
 |
747 | YDB(YDBError),
 | ^^^ help: consider making the acronym lowercase, except the initial letter: `Ydb`
 |
 = note: `-D clippy::upper-case-acronyms` implied by `-D clippy::all`
 = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms
```
